DetoxSync/DetoxSync/Spies/NSTimer+DTXSpy.m

```diff
@@ -94,9 +97,9 @@ void __detox_sync_CFRunLoopRemoveTimer(CFRunLoopRef rl, CFRunLoopTimerRef timer,
 //	NSLog(@"🤦‍♂️ removeTimer: %@", NS(timer));
 	
 	id<DTXTimerProxy> trampoline = [DTXTimerSyncResource existingTimerProxyWithTimer:NS(timer)];
-	[trampoline untrack];
-	
 	__orig_CFRunLoopRemoveTimer(rl, timer, mode);
+
+	[trampoline untrack];
```

Comment:
I've observed that `[_DTXTimerTrampoline fire:]` is called after the trampoline has been untracked (and therefore `dealloc`ed).

By switching these, I was hoping that the timer would be cancelled (ensuring that the `fire` method is not called after untracking).

This doesn't seem to be the case, and I'm still seeing `fire` called after the trampoline has been untracked, but it still feels safer to me to remove the timer before untracking?

Reply:
Untracking the timer only after removing it from the runloop does make sense, but I'm not sure if it matters. 

> [_DTXTimerTrampoline fire:] is called after the trampoline has been untracked

That's very weird.

Reply:
> [_DTXTimerTrampoline fire:] is called after the trampoline has been untracked

That's very weird. 🤔

Reply:
Maybe the thread where this timer was created and the thread where the timer was called to invalidate are different?

Apple docs ([invalidate()](https://developer.apple.com/documentation/foundation/timer/1415405-invalidate)):
>You must send this message from the thread on which the timer was installed. If you send this message from another thread, the input source associated with the timer may not be removed from its run loop, which could prevent the thread from exiting properly.